From b672507ca9f06bb17213036b16bc4f5c5bc65357 Mon Sep 17 00:00:00 2001
From: Shiji Yang <yangshiji66@outlook.com>
Date: Sun, 22 Dec 2024 17:06:59 +0800
Subject: [PATCH 3/4] rt2x00: correct MT7620 SDM mode register value

rt2x00_set_field8() is a mask writing function. If we want to set
the BIT(7) for the SDM mode register here, we only need to fill "4"
in the mask.

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
---
 drivers/net/wireless/ralink/rt2x00/rt2800lib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/ralink/rt2x00/rt2800lib.c
+++ b/drivers/net/wireless/ralink/rt2x00/rt2800lib.c
@@ -3848,7 +3848,7 @@ static void rt2800_config_channel_rf7620
 
 	/* Default: XO=20MHz , SDM mode */
 	rfcsr = rt2800_rfcsr_read(rt2x00dev, 16);
-	rt2x00_set_field8(&rfcsr, RFCSR16_SDM_MODE_MT7620, 0x80);
+	rt2x00_set_field8(&rfcsr, RFCSR16_SDM_MODE_MT7620, 4);
 	rt2800_rfcsr_write(rt2x00dev, 16, rfcsr);
 
 	rfcsr = rt2800_rfcsr_read(rt2x00dev, 21);
